### ISSUE TITLE
Getting `failure: dial attempt failed: context deadline exceeded` on macos

### ISSUE THREAD

**Issue by dryajov** [libp2p/go-libp2p-circuit#27]
#### Version information:

```
go-ipfs version: 0.4.13-
Repo version: 6
System version: amd64/darwin
Golang version: go1.9.2
```

#### Description:

Getting `failure: dial attempt failed: context deadline exceeded` when dialing over go circuit to either a js or a go node. This only happens on MacOS, it works fine on both linux and windows platforms. 

To reproduce, run the interop tests in https://github.com/ipfs/interop/pull/6 on macos.

```
  1) circuit go relay go <-> go relay <-> go should connect:
     Error: connect QmPESBd38krVFxtykakWtNzNTn9wAfGDry9nkjSzYcJe1R failure: dial attempt failed: context deadline exceeded
      at parseError (node_modules/ipfs-api/src/utils/send-request.js:16:17)
      at ClientRequest.<anonymous> (node_modules/ipfs-api/src/utils/send-request.js:39:14)
      at HTTPParser.parserOnIncomingClient (_http_client.js:551:21)
      at HTTPParser.parserOnHeadersComplete (_http_common.js:117:23)
      at Socket.socketOnData (_http_client.js:440:20)
      at addChunk (_stream_readable.js:263:12)
      at readableAddChunk (_stream_readable.js:250:11)
      at Socket.Readable.push (_stream_readable.js:208:10)
      at TCP.onread (net.js:594:20)

  2) circuit go relay go <-> go relay <-> go should transfer:
     Error: Timeout of 80000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
 
```

**Comment by dryajov** [comment]
I was able to confirm that branch https://github.com/ipfs/go-ipfs/tree/gx/update-1m7019 fixes the above issues.

**Comment by Stebalien** [comment]
Will be fixed once ipfs/go-ipfs#4610 is merged.

**Comment by Stebalien** [comment]
Fixed.
